lib/Sema/TypeCheckDecl.cpp

```diff
@@ -482,6 +482,53 @@ static void checkInheritanceClause(
   }
 }
 
+// Check for static properties that produce empty option sets
+static void checkForEmptyOptionSet(const VarDecl *VD, TypeChecker &tc) {
+  if (!VD->isStatic() || !VD->isLet())
+    return;
+  
+  auto DC = VD->getDeclContext();
+  
+  if (!VD->getType()->isEqual(DC->getSelfTypeInContext()))
+    return;
+  
+  auto *optionSetProto = tc.Context.getProtocol(KnownProtocolKind::OptionSet);
+  auto protocolConformance = !tc.containsProtocol(
+                                                  DC->getSelfTypeInContext(),
+                                                  optionSetProto,
+                                                  DC,
+                                                  /*Flags*/None);
+  
+  if (!protocolConformance)
```

Comment:
```suggestion
  // Make sure this type conforms to OptionSet
  if (!protocolConformance)
```

Reply:
For that matter, it might make sense to name the variable `optionSetConformance`—it's more descriptive.